src/main/java/com/aws/greengrass/mqtt/bridge/auth/MQTTClientKeyStore.java

```diff
@@ -132,6 +137,42 @@ public void updateCA(List<String> caCerts) throws IOException, CertificateExcept
         updateListeners.forEach(UpdateListener::onCAUpdate); //notify MQTTClient
     }
 
+    /**
+     * Get the CA cert chain from the key store.
+     *
+     * @return ca cert chain
+     * @throws KeyStoreException if keystore has not been initialized
+     * @throws CertificateEncodingException if an encoding error occurs
+     */
+    @SuppressWarnings("PMD.InsufficientStringBufferDeclaration")
+    public Optional<String> getCACertChain() throws KeyStoreException, CertificateEncodingException {
```

Comment:
We already have a list of the CAs as strings: https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/22f37e3a4eee4172b5c222c06be56d48935b16b7/src/main/java/com/aws/greengrass/mqtt/bridge/MQTTBridge.java#L188.

Let's not convert back and forth when it isn't needed.

Reply:
Storing the list in a field in this class is fine too.

Reply:
good point, will fix

Reply:
CDA only ever returns chain of length 1? 

https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/1016cb5ebc0fef7f959f4df00aec784bd316e9b4/src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java#L148

usages here: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/1016cb5ebc0fef7f959f4df00aec784bd316e9b4/src/main/java/com/aws/greengrass/clientdevices/auth/certificate/usecases/ConfigureCustomCertificateAuthority.java#L56

and here: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/1016cb5ebc0fef7f959f4df00aec784bd316e9b4/src/main/java/com/aws/greengrass/clientdevices/auth/certificate/usecases/ConfigureManagedCertificateAuthority.java#L60

Reply:
more of a curiosity, will still make it adaptable to full chain

Reply:
https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/cd64f652effad0de98f88dd2204ddc9e4e59e6f0/src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java#L82-L83